frontend/src/lib/lemon-ui/LemonTree/LemonTree.tsx

```diff
@@ -950,11 +1012,30 @@ const LemonTree = forwardRef<LemonTreeRef, LemonTreeProps>(
             [expandedItemIdsState, onFolderClick, onNodeClick, focusContent]
         )
 
+        /** Focus the element. */
+        const focusElement = useCallback((element: HTMLElement) => {
+            if (element) {
+                // Focus the element
+                element.focus()
+            }
+        }, [])
+
+        /** Focus the element from the tree item ID. */
+        const focusElementFromId = useCallback((id: string) => {
+            // Timeout to ensure the element is rendered
+            setTimeout(() => {
+                // Now use the escaped ID in your query
+                const element = containerRef.current?.querySelector(`[data-id="${escapeId(id)}"]`) as HTMLElement
+                // Focus the element
+                focusElement(element)
```

Comment:
I'm not too sure of the focusElement callback either. You could just go `element?.focus()`. If you do want these to be separate functions, then as they have no other variables than inputs, hoist them to the global scope.